### PR TITLE
che-docs #715 Adding documentation for Hosted Che

### PR DESCRIPTION
**Pull request by ibuziuk**
Signed-off-by: Ilya Buziuk <ibuziuk@redhat.com>

### What does this PR do?
Adding documentation for Hosted Che

### What issues does this PR fix or reference?
#715

### CODE REVIEW

File: src/main/pages/che-7/overview/assembly_hosted-che.adoc
Comment (themr0c) [inline src/main/pages/che-7/overview/assembly_hosted-che.adoc]: Hi @ibuziuk, such a long list of screenshot will be difficult to maintain.
Websites should be documented by saying click on the Blah button.
Maybe a screenshot for the result.
Would you mind replacing the screenshots by a more textual description?

Comment (themr0c) [reply src/main/pages/che-7/overview/assembly_hosted-che.adoc]:
```suggestion
. Navigate to https://che.openshift.io/.
. Log in with your existing OpenShift Online, Red Hat Developer Program, or Red Hat Customer Portal account, or register for a Red Hat account if you don't have one already.
. In the next screen, activate your account by clicking on the *Activate account* button.
. Your environment is being built and then complete, you will receive an email with a link to connect to your hosted Eclipse Che instance.
. Follow the link and the email and start to use Eclipse Che by clicking on the *Get started* button.
```

File: src/main/pages/che-7/overview/assembly_hosted-che.adoc
Comment (themr0c) [inline src/main/pages/che-7/overview/assembly_hosted-che.adoc]:
```suggestion
If you need to access code on GitHub, it is recommended to link your GitHub account to your Red Hat account.

. Navigate to https://developers.redhat.com/auth/realms/rhd/account/identity
. Click on the *Link* button next to the GitHub logo.

```

File: src/main/pages/che-7/overview/assembly_hosted-che.adoc
Comment (themr0c) [inline src/main/pages/che-7/overview/assembly_hosted-che.adoc]:
```suggestion
== Linking your GitHub Account
```

File: src/main/pages/che-7/overview/assembly_hosted-che.adoc
Comment (themr0c) [inline src/main/pages/che-7/overview/assembly_hosted-che.adoc]:
```suggestion
== Registering to Hosted Che
```

File: src/main/pages/che-7/overview/assembly_hosted-che.adoc
Comment (themr0c) [inline src/main/pages/che-7/overview/assembly_hosted-che.adoc]:
```suggestion
=== Create a Che 7 Workspace using a wizard

Follow the wizard to create a workspace from a template or from your GitHub or GitLab repository.

. Navigate to https://www.eclipse.org/che/getting-started/cloud/
. Choose the desired flavor and launch your new Workspace.
```

**Review by themr0c — COMMENTED** [review]
Hello @ibuziuk, what do you think of the proposed suggestions?

File: src/main/pages/che-7/overview/assembly_hosted-che.adoc
Comment (ibuziuk) [reply src/main/pages/che-7/overview/assembly_hosted-che.adoc]: Good point, but I believe we would like to promote account linking via Che User Dashboard. This way of linking could be treated as optional

**Comment by ibuziuk** [comment]
@rkratky @supriya-bharadwaj @themr0c PR is ready for review

File: src/main/pages/che-7/overview/assembly_hosted-che.adoc
Comment (l0rd) [inline src/main/pages/che-7/overview/assembly_hosted-che.adoc]: There should be a section dedicated to this topic (c.f. [the section in che-docs](https://github.com/eclipse/che-docs/blob/master/src/main/pages/che-7/hands-on-guidance/proc_configuring-your-vcs-credentials-for-your-workspaces.adoc) and [the google doc for the draft](https://docs.google.com/document/d/15OGTxr1ChL74DcBMaiUeL0QKJqdPHn2HnTm8WRtrn7k/) ). I would rather add a link to that doc here.

File: src/main/pages/che-7/overview/assembly_hosted-che.adoc
Comment (l0rd) [inline src/main/pages/che-7/overview/assembly_hosted-che.adoc]: We are centralizing all the issues/README/wiki to eclipse/che. Wouldn't make sense to move the hosted-che FAQ to eclipse/che as well?

**Review by l0rd — APPROVED** [review]
Really really good. Concise and clear. I have added a couple of comment but these are really minor stuff so I am ok to merge even with the current PR content.

File: src/main/pages/che-7/overview/assembly_hosted-che.adoc
Comment (ibuziuk) [reply src/main/pages/che-7/overview/assembly_hosted-che.adoc]: good point

File: src/main/pages/che-7/overview/assembly_hosted-che.adoc
Comment (ibuziuk) [reply src/main/pages/che-7/overview/assembly_hosted-che.adoc]: good point :+1:

**Comment by ibuziuk** [comment]
@l0rd thank you for review - have added a link to "Configuring your VCS credentials for your workspaces" + moved FAQ / Troubleshooting to the doc. Once PR is merged and Hosted Che docs will be available we would also need to update the link in the UD footer - https://github.com/redhat-developer/rh-che/issues/1479

**Review by Katka92 — APPROVED** [review]
Good job! :+1:

**Comment by themr0c** [comment]
@l0rd What is exactly the plan for the Hosted Che FAQ? 

- regroup it with the current Hosted Che doc in the repository https://github.com/eclipse/che-docs to make it available in https://www.eclipse.org/che/docs/che-7/index.html. In other word we need to integrate here all FAQ entries currently in https://github.com/redhat-developer/rh-che/blob/master/FAQ.adoc.
- regroup it with issues/README/wiki to the repository in https://github.com/eclipse/che. In other words remove FAQ entries from current PR?

**Comment by ibuziuk** [comment]
I will let @l0rd to comment but for me it does not make sense to Have Hosted Che docs in the  https://github.com/eclipse/che wiki. I personally would really like to have all the Hosted Che docs consolidated in one place (including FAQ) and get rid of the https://github.com/redhat-developer/rh-che/blob/master/FAQ.adoc from the Downstream repo.

**Comment by l0rd** [comment]
I think there should be consistency. If we have upstream Che FAQ in the `eclipse/che` wiki we should have hosted Che FAQ in that same wiki as well. Or we can move both in che-docs. But it should be consistent.

**Comment by rkratky** [comment]
@l0rd, I agree about consistency, but I think that what you're suggesting actually makes things more confusing and inconsistent. What would make sense from the perspective of a user who doesn't know about our internal discussions and processes, is to have an FAQ doc in the repo to which it belongs. Putting Hosted Che FAQ to the upstream Eclipse Che repo is not intuitive.

Upstream Che is an Eclipse Foundation project. Hosted Che is a Red Hat product. Just as we don't put CRW docs into the Eclipse Che repo, we shouldn't, in my opinion, do it with Hosted Che.

**Comment by ibuziuk** [comment]
> Upstream Che is an Eclipse Foundation project. Hosted Che is a Red Hat product. Just as we don't put CRW docs into the Eclipse Che repo, we shouldn't, in my opinion, do it with Hosted Che.

I also think that putting Hosted Che FAQ [1] to upstream would be more confusing. However, Hosted Che FAQ bits are not part of the Che 7 GA endgame plan. So, I would propose for now:

- leave Hosted Che FAQ in the rh-che repo [1] as it was originally implemented as part of the https://github.com/redhat-developer/rh-che/issues/1287
- In the Hosted Che docs have a link to this FAQ as was originally done in this PR:

````
[id='faq']
== Frequently Asked Questions (FAQs) and Troubleshooting
Frequently Asked Questions (FAQs) and Troubleshooting can be found on the following link:https://github.com/redhat-developer/rh-che/blob/master/FAQ.adoc[link].
````

After GA release we could reconsider what would be the best place for FAQ. 
@rkratky @l0rd wdyt?


[1] https://github.com/redhat-developer/rh-che/blob/master/FAQ.adoc

**Comment by rkratky** [comment]
+1 @ibuziuk

**Comment by l0rd** [comment]
Currently on Che community website there are direct links to hosted Che.

On Che community documentation there are direct links to hosted Che as well.

If we want to be consistent with what you are proposing we should move
those links away from upstream because, as you said, hosted Che is a Red
Hat product and we should separate community and downstream or user and
contributors will be confused.

However we are pushing for presenting hosted Che as the place to go and try
upstream Che (not CodeReady Workspaces). We are pushing to make upstream
and hosted Che based on the same identical (community) codebase. We are
considering deprecating rh-che GitHub repository.

I believe the latter is the right direction for us today (unification). But
I understand and find reasonable the first approach as well (separation).
But on both cases we need to be consistent.

**Comment by l0rd** [comment]
Anyway +1 to merge this PR as it is and open an issue/question to discuss where the hosted-che FAQ should be located.

**Comment by ibuziuk** [comment]
@l0rd sounds good, I have updated FAQ related issue https://github.com/redhat-developer/rh-che/issues/1479 accordingly

**Review by themr0c — COMMENTED** [review]
I aggregated suggestions in https://github.com/ibuziuk/che-docs/pull/1/files, I will remove my flood and suggestions from here, and I apologize for the flood.

**Comment by ibuziuk** [comment]
@rkratky will let you review first ;-) ^

**Comment by ibuziuk** [comment]
@rkratky @themr0c is it expected that all the images but one  `getting-started-devfile.png` have been removed ?

**Comment by ibuziuk** [comment]
````
Creating a worskpace from the user dashboard
This section describes how to create a workspace from the user dashboard in Hosted Che.

Prerequisites
Existing Hosted Che account. See Registering to Hosted Che.

````
Link `Registering to Hosted Che.` is broken

**Comment by themr0c** [comment]
@ibuziuk Yes I removed most images and replaced them with a textual description which should last longer.

**Comment by ibuziuk** [comment]
@themr0c most = all but one ?

**Comment by themr0c** [comment]
Yes, screenshots that are at the end of the procedure are fine.

**Comment by ibuziuk** [comment]
@themr0c PR is  https://github.com/ibuziuk/che-docs/pull/1/files merged and all the changes are squashed in 1 commit.

@rkratky waiting for your final approval before merging to master

**Review by themr0c — APPROVED** [review]

**Review by rkratky — APPROVED** [review]

**Pull request merged**
